src/app/diagnostics/logger.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from "@angular/core";
+
+export enum LogLevel {
+    DEBUG = 0,
+    INFO,
+    WARN,
+    ERROR,
+}
+
+export interface Logger {
+    log(level: LogLevel, msg: string)
+}
+
+class ConsoleLogger implements Logger {
+    constructor(
+        private senderType: string,
+    ) {}
+
+    public log(level: LogLevel, message: string) {
+        var formattedMsg = `(${this.senderType}): ${message}`
+        switch(level) {
+            case LogLevel.DEBUG:
+                console.debug(formattedMsg)
+                break
+
+            case LogLevel.INFO:
+                console.info(formattedMsg)
+                break
+
+            case LogLevel.WARN:
+                console.warn(formattedMsg)
+                break
+
+            case LogLevel.ERROR:
+                console.error(formattedMsg)
```

Comment:
Can we set different color for error like "console.log(formattedMsg, 'background: #222; color: #bada55');"?

FYI, https://stackoverflow.com/questions/7505623/colors-in-javascript-console

Reply:
This sounds like a good idea. I would like to put more thought into the color though. Since it custom color is a extra feature, I would like to delay the changes for now. Let's talk tomorrow when you are in and we will create a new PR